Looks like you need to make a small adjustment:
```
/tmp/cirrus-ci-build/libarchive/archive_read_support_format_rar.c/tmp/cirrus-ci-build/libarchive/archive_read_support_format_rar.c:2552:28: error: comparison of integers of different signs: 'int' and 'unsigned int' [-Werror,-Wsign-compare]
 2552 |       (rar->lzss.mask + 1) < rar->dictionary_size)
      |        ~~~~~~~~~~~~~~~~~~  ^ ~~~~~~~~~~~~~~~~~~~~
:2552:28: error: comparison of integers of different signs: 'int' and 'unsigned int' [-Werror,-Wsign-compare]
 2552 |       (rar->lzss.mask + 1) < rar->dictionary_size)
      |        ~~~~~~~~~~~~~~~~~~  ^ ~~~~~~~~~~~~~~~~~~~~
```

--- a/libarchive/archive_read_support_format_rar.c
+++ b/libarchive/archive_read_support_format_rar.c
@@ -2549,7 +2549,7 @@ parse_codes(struct archive_read *a)
   }
 
   if (!rar->dictionary_size || !rar->lzss.window ||
-      (rar->lzss.mask + 1) < rar->dictionary_size)
+      (unsigned int)(rar->lzss.mask + 1) < rar->dictionary_size)
   {
     /* Seems as though dictionary sizes are not used. Even so, minimize
      * memory usage as much as possible.

